sdk/Sdk/FunctionMetadataGenerator.cs

```diff
@@ -98,17 +98,24 @@ public IEnumerable<SdkFunctionMetadata> GenerateFunctionMetadata(string assembly
         internal IEnumerable<SdkFunctionMetadata> GenerateFunctionMetadata(ModuleDefinition module)
         {
             var functions = new List<SdkFunctionMetadata>();
+            bool functionFound = false;
 
             foreach (TypeDefinition type in module.Types)
             {
                 var functionsResult = GenerateFunctionMetadata(type).ToArray();
                 if (functionsResult.Any())
                 {
+                    functionFound = true;
                     _logger.LogMessage($"Found {functionsResult.Length} functions in '{type.GetReflectionFullName()}'.");
                 }
 
                 functions.AddRange(functionsResult);
             }
+            
+            if (!functionFound && TryAddExtensionInfo(_extensions, module.Assembly, usedByFunction: false))
```

Comment:
If `functionFound` is `true` , we are not adding the implicit registration thing ?

Reply:
Wonder if it would be clearer (if it would have avoided Shyju's question) if the variable was named something like `moduleExtensionsRegistered` and perhaps a comment down where we evaluate describing that modules with defined functions will have their extension references registered.